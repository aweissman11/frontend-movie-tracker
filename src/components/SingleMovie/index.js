import React from 'react';
import PropTypes from 'prop-types';

import FavoriteBtn from '../../containers/FavoriteBtn';

const SingleMovie = (props) => {
  const date = props.release_date;
  const year = date.split('').splice(0, 4).join('');
  return (
  <div className='single-movie-wrapper'>
    <div className='favorite-btn-wrapper'>
      <FavoriteBtn movieId={props.id}/>
    </div>
    {
      props.poster_path ?
    <img
      src={`https://image.tmdb.org/t/p/w400_and_h600_bestv2${props.poster_path}`} 
      alt={props.title}
      /> :
    <img
      src={'http://www.beguilingbooksandart.com/wp-content/uploads/2015/01/at_marcy.png'} 
      alt={props.title}
      />
    }
    <section className='movie-poster-text'>
      <h2>{props.title}</h2>
      <h3>{year}</h3>
    </section>
  </div>
)}


SingleMovie.propTypes = {
  date: PropTypes.object,
  year: PropTypes.object
};

export default SingleMovie;