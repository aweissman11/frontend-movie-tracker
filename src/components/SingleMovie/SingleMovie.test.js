import React from 'react';
import SingleMovie from './';
import { shallow } from 'enzyme';

describe('SingleMovie', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<SingleMovie
      title='test-title'
      release_date='2018-01-01'
      id='100'
      poster_path='test_path'
    />);
    
    expect(wrapper).toMatchSnapshot()
  });
});

