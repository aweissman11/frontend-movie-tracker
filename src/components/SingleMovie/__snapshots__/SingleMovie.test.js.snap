// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleMovie should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SingleMovie />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2 />,
        <Connect(FavoriteBtn) />,
        <img
          src="https://image.tmdb.org/t/p/w400_and_h600_bestv2undefined"
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "movieId": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": undefined,
          "src": "https://image.tmdb.org/t/p/w400_and_h600_bestv2undefined",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2 />,
          <Connect(FavoriteBtn) />,
          <img
            src="https://image.tmdb.org/t/p/w400_and_h600_bestv2undefined"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "movieId": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": undefined,
            "src": "https://image.tmdb.org/t/p/w400_and_h600_bestv2undefined",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
